python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -385,25 +386,32 @@ def generate_session_token(hostname, login, password, http_proxy):
     :param login: The user to get a session for.
     :param password: Password for the user.
     :param http_proxy: Proxy to use. Can be None.
+    :param auth_token: Two factor authentication token for the user. Can be None.
 
-    :returns: The generated session token for that user/password/site combo.
+    :returns: The generated session token for that user/password/site combo. If the login
```

Comment:
ok, without knowing much about the code at all, my opinion on exceptions in this type of scenario is this:  If the code can handle the exception silently and not leave the calling code in a state where it doesn't know what happened then by all means swallow the exception.  If the calling code may need to know what happened then the exception should be raised.  Also, if returning 'None' is an error condition stemming from an exception then why not just raise the exception.  Looking at this method in complete isolation, it is odd that this particular exception is swallowed when all others are raised - the method is essentially making an assumption about how it is going to be used but this isn't clear in either the method name, or arguments and breaks encapsulation - it's also not clear why it is doing this - why is suppressing this exception a good idea?  If this behaviour is needed then maybe adding an extra argument to the call would make it more explicit, something like:  

def generate_session_token(hostname, login, password, http_proxy, auth_token=None, ignore_missing_2fa=True)

don't know if that helps at all ;)